### DISCUSSION TITLE
Open system monitor with a right click on the desktop

### DISCUSSION THREAD

**Discussion by 100savage** [linuxmint/discussions#233] [Tutorials]
Open System Monitor with a simple right-click on the desktop.

Create a file called **monitor.nemo_action** and place it in your **.local/share/nemo/actions/**   directory
Note: the .local folder is hidden and located in your home directory.

```
[Nemo Action]
Name=System Monitor
Comment=View current processes and monitor system state
Exec=gnome-system-monitor
Icon-Name=org.gnome.SystemMonitor
Selection=None
Extensions=any;
Conditions=desktop;
```
Open a terminal and type the below command to restart nemo
`nemo -q`

Right-click on your desktop and select System Monitor

**Comment by rcalixte** [comment]
Would you like to submit this as a new action [here](https://github.com/linuxmint/cinnamon-spices-actions)?

This would offer several benefits:
* It would make it easier to distribute to users rather than relying on copying, pasting, and commands. (I'm also not sure if the Exec statement needs the %F parameter since it's just calling the binary.)
* Support could be added for translations.

**Reply by 100savage** [reply]
Thanks
You are correct, the %F is not needed. I updated the file and submitted the new action request.

**Reply by rcalixte** [reply]
Feel free to take a look at the README or the other existing actions for how to structure your new submission. Or if you're okay being listed as the maintainer, I can help push your action through.

**Reply by 100savage** [reply]
If you don't mind, please push my action through. I'm okay with being listed as the maintainer.
Thanks

**Comment by anaximeno** [comment]
@100savage please consider the option suggested above to create a new action. It'll work similar to other spices (applets, desklets, etc), and will be easy for users to install them on their system!

**Comment by rcalixte** [comment]
This is now live [here](https://cinnamon-spices.linuxmint.com/actions/view/6). Thank you @100savage!

**Comment by MartinHaeusler** [comment]
I would like to add another related request to this: can we make the system monitor available as a menu item for the task bar context menu? So, right-click on an empty spot in the task bar -> System Monitor. Gnome uses this pattern, Windows as well. I find myself looking for that quite a lot actually. It's a papercut, but one that cuts me often.

**Reply by rcalixte** [reply]
I believe you are referring to the panel. That context menu is a different code path than what Actions can control or modify. You can certainly create a new feature request but I don't know what the appetite will be for that change since it would be a global change affecting all users (even users who have something other than GNOME System Monitor installed).
